controllers/cardsController.js

```diff
@@ -69,43 +69,74 @@ exports.cardDetail = function(req, res, next) {
 				var err = new Error('Card not found');
 				return next(err);
 			}
-		} else {
-			hasCard = false;
-			if (req.user) {
-				var [err, hasCard] = await isCardInCollection(req.user.name, req.params.id);
-				if (err) { return next(err); }
+		} else { // we found the card
+			var stats = new Object();
+			if (req.user) { // check if user have card
+				try {
+					if (JSON.stringify(await Users.find({_id: req.user.id, "cards.owned": req.params.id})) === "[]") {
```

Comment:
Can be simplified as `if (await Users.exists({_id: req.user.id, "cards.owned": req.params.id}))`